File: src/SSHServer/ExchangeContext.cs
using SSHServer.Ciphers;
using SSHServer.Compressions;
using SSHServer.HostKeyAlgorithms;
using SSHServer.KexAlgorithms;
using SSHServer.MACAlgorithms;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace SSHServer
{
    public class ExchangeContext
    {
        public IKexAlgorithm KexAlgorithm { get; set; } = null;
        public IHostKeyAlgorithm HostKeyAlgorithm { get; set; } = null;
        public ICipher CipherClientToServer { get; set; } = new NoCipher();
        public ICipher CipherServerToClient { get; set; } = new NoCipher();
        public IMACAlgorithm MACAlgorithmClientToServer { get; set; } = null;
        public IMACAlgorithm MACAlgorithmServerToClient { get; set; } = null;
        public ICompression CompressionClientToServer { get; set; } = new NoCompression();
        public ICompression CompressionServerToClient { get; set; } = new NoCompression();
    }
}

File: SSHServerNodeJS/SSHServerNodeJS/ExchangeContext.ts
import { IKexAlgorithm } from "./KexAlgorithms/IKexAlgorithm";
import { IHostKeyAlgorithm } from "./HostKeyAlgorithms/IHostKeyAlgorithm";
import { ICipher } from "./Ciphers/ICipher";
import { NoCipher } from "./Ciphers/NoCipher";
import { IMACAlgorithm } from "./MACAlgorithms/IMACAlgorithm";
import { ICompression } from "./Compressions/ICompression";
import { NoCompression } from "./Compressions/NoCompression";

export class ExchangeContext {
    public kexAlgorithm: IKexAlgorithm = null;
    public hostKeyAlgorithm: IHostKeyAlgorithm = null;
    public cipherClientToServer: ICipher = new NoCipher();
    public cipherServerToClient: ICipher = new NoCipher();
    public macAlgorithmClientToServer: IMACAlgorithm = null;
    public macAlgorithmServerToClient: IMACAlgorithm = null;
    public compressionClientToServer: ICompression = new NoCompression();
    public compressionServerToClient: ICompression = new NoCompression();
}
